iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/StringLiteralTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.docker.tree.impl;
+
+import org.junit.jupiter.api.Test;
+import org.sonar.iac.docker.parser.grammar.DockerLexicalGrammar;
+import org.sonar.iac.docker.parser.utils.Assertions;
+
+class StringLiteralTest {
+
+  @Test
+  void test() {
+    Assertions.assertThat(DockerLexicalGrammar.STRING_LITERAL)
+      .matches("f")
+      .matches("foo")
+      .matches("   foo")
+      .matches("1")
+      .matches("123")
+      .matches("SIGKILL");
+  }
```

Comment:
Good idea for putting test on the `STRING_LITERAL` !
I would add few more use case, especially about `STRING_LITERAL_WITH_QUOTES` versions :
`"\"mystring\""`
`"\"partial_quotes_1"`
`"partial_quotes_2\""`
` ` (empty)